python/ray/dag/dag_node_operation.py

```diff
@@ -195,6 +189,30 @@ def _add_edge(from_node: _DAGOperationGraphNode, to_node: _DAGOperationGraphNode
     to_node.in_edges.add((from_node.task_idx, from_node.operation.type))
 
 
+def _update_candidate_nodes(
+    actor_to_candidates: Dict["ray._raylet.ActorID", List[_DAGOperationGraphNode]],
+    graph: Dict[int, Dict[_DAGNodeOperationType, _DAGOperationGraphNode]],
+    node: _DAGOperationGraphNode,
+) -> None:
+    # There is only one NCCL collective node in the collective operation that
+    # is added to the candidates. The collective operation is ready when all
+    # the nodes have zero in-degrees. Except for the last node, all the other
+    # nodes are not ready yet. When the last node is selected, all the nodes
+    # in the collective operation are returned.
```

Comment:
```suggestion
    # Collective operations are ready when all
    # the nodes in the collective have zero in-degrees.
    # Only one node per collective will be added as ready.
```

(comments should not reference logic in other functions)

Reply:
Done.